IDR/notebook/scripts/nbconverted/1.describe_screens.py

```diff
@@ -0,0 +1,317 @@
+
+    details_ = (
+        pd.DataFrame({x[0]: x[1]
+                      for x in annotations[study_index]["values"]}, index=[0])
+        .assign(
+            internal_id=id_,
+            upload_date=date_,
+            idr_name=name_,
+            screen_id=screen_id
+        )
+    )
+
+    return details_
+
+
+def describe_screen(screen_id):
+    """Pull additional metadata info per plate, given screen id
+
+    Parameters
+    ----------
+    screen_id: int
+        ID of the screen data set
+
+    Returns
+    -------
+    pandas.DataFrame() of the following metadata values:
+
+        screen_id: IDR ID for the screen
+        plate_id: IDR ID for each plate
+        plate_name: Names given to each plate
+        image_id: IDR ID for each image
+        cell_line: Cell line used in the screen experiment
+        strain: Strain of the cell line (if specified)
+        gene_identifier: Accession code for the gene being perturbed in a well
+        phenotype_identifier: Accession code for the phenotype perturbed in a well
+        stain: Set of the stains used in the screen
+        stain_target: The target protein or media of the stain
+        pixel_size_x: Width of the image
+        pixel_size_y: Height of the image
+        imaging_method: Method used to collect images (ex: fluorescence microscopy)
+    """
+    session = requests.Session()
+
+    # Get number of plates per screen and append to a dictionary
+    PLATES_URL = f"https://idr.openmicroscopy.org/webclient/api/plates/?id={screen_id}"
+    all_plates = session.get(PLATES_URL).json()["plates"]
+    study_plates = {x["id"]: x["name"] for x in all_plates}
+    print(f"Number of plates found in screen {screen_id}: ", len(study_plates))
+
+    # Initialize results list
+    plate_results = list()
+
+    # Iterate through all plates in the study
+    for plate in study_plates:
+        imageIDs = list()
+        plate_name = study_plates[plate]
+
+        # Access .json file for the plate ID. Contains image ID (id) numbers
+        # for replicate images per plate.
+        WELLS_IMAGES_URL = f"https://idr.openmicroscopy.org/webgateway/plate/{plate}/"
+        grid = session.get(WELLS_IMAGES_URL).json()
+
+        try:
+            pixel_size_x = grid["image_sizes"][0]["x"]
+            pixel_size_y = grid["image_sizes"][0]["y"]
+
+            # Get image and well ids
+            for image in grid['grid'][0]:
+                # Append IDs to iterable lists
+                thumb_url = image['thumb_url'].rstrip(image['thumb_url'][-1])
+                image_id = thumb_url.split('/')[-1]
+                imageIDs.append(image_id)
+
+        except (ValueError, KeyError):
+            plate_results.append(
+                [screen_id, plate, plate_name, None, None, None, None, None])
+            continue
+
+        # TODO: Rewrite to optimize and make consice
+        # TODO: Include cell/tissue label and map IDR accession to each screen
```

Comment:
Looks like some TODOs snuck in. Please make sure to run the same `nbconvert` command when notebooks update.

Reply:
I regenerated the converted notebook files to include changes made during this PR using `nbconvert` command.

Added in d165a301eb7222601fff5ce52254c2999a800f47.